scripts/algorithms/common/helper_functions.py

```diff
@@ -32,3 +34,46 @@ def set_random_seed(seed: int, env: gym.Env):
     torch.manual_seed(seed)
     np.random.seed(seed)
     random.seed(seed)
+
+
+def get_n_step_info_from_demo(
+    demo: List, n_step: int, gamma: float
+) -> Tuple[List, List]:
+    """Return 1 step and n step demos."""
+    assert demo
+    assert n_step > 1
+
+    demos_1_step = list()
+    demos_n_step = list()
+    n_step_buffer: Deque = deque(maxlen=n_step)
+
+    for transition in demo:
+        n_step_buffer.append(transition)
+
+        if len(n_step_buffer) == n_step:
+            # add a single step transition
+            demos_1_step.append(n_step_buffer[0])
+
+            # add a multi step transition
+            curr_state, action = transition[:2]
```

Comment:
Sorry. My mistake.
This line should be: `curr_state, action = self.n_step_buffer[0][:2]`

Reply:
Oh that makes sense! Since we append from the right, the first one is the state N steps before. Pushed commit 33bae1e that fixes this issue.